src/core/toga/widgets/canvas.py

```diff
@@ -591,6 +646,32 @@ def _draw(self, impl, *args, **kwargs):
         impl.move_to(self.x, self.y, *args, **kwargs)
 
 
+class CloseTo:
+    """A user-created :class:`CloseTo <CloseTo>` drawing object which closes
+     the polyline vack to the starting point
```

Comment:
I'm a little unclear about why CloseTo is needed if we also have ClosePath. Is there really a need for two different ways to draw a closed path?

Reply:
with close path you have to know before adding the path segments that it is closed. matplotlib gives an array of draw commands. When I loop over the commands, I do not know it is closed till that last command telling me to close the segment.  With the two implemented platforms so far. To close the segment does not require passing it the beginning point to close to. So adding this type makes it easier for me to just loop over the draw commands and add each one to the drawing context.

Reply:
Ok; that makes some sense; I missed that this isn't a context, it's a drawing primitive. 

I *think* I understand the use case; however, I'm still not 100% sure if a new primitive is needed here.

As I understand it, the problem is that at the time you are drawing the primitives, you don't know if the path is closed or open, so you need to be able to put a "close" command at the end of sequence of drawing commands.

My countersuggestion would be that you could use the ClosedPath context manager in a different way to acheive the same goal.

ClosedPath is a context manager, so the default way of using it is:
```
with canvas.closed_path(x,y) as path:
    path.line_to(x+10, y+10)
    path.line_to(x-10, y+10)
```
but that's no the only way it can be used. A context manager is just an object that has an enter/exit lifecycle; you can create instances of the object and call the underlying primitives on it directly. So, if you don't know if the path is closed until some boolean is evaluated, you could do something like:
```
draw_objs = [
    LineTo(x+10, y+10),
    LineTo(x-10, y+10),
]

if is_closed:
    target = canvas.closed_path(x,y)
else:
    target = canvas

for obj in draw_objs:
    target.add_draw_obj(obj)
```
That last part could be simplified by adding an "add_draw_objs()" method, which seems like a likely use case.

The underlying idea here is that closing a path isn't an operation; a closed path is an entity that is constructed from other operations. Does that make sense?

(Also, small typo in the docstring - `vack`)